### PR TITLE
Rebase with Upstream

### PR DESCRIPTION
**Pull request by alecmerdler**
# What Are We Doing Here

Rebased upstream changes.

## How to Verify

1. Check out this PR
2. Run `make clean && make build`
3. ???
4. Profit

### CODE REVIEW

File: Gopkg.lock [hunk 1]
```diff
@@ -1,1314 +0,0 @@
-# This file is autogenerated, do not edit; changes may be undone by the next 'dep ensure'.
```
Comment (ecordell) [inline Gopkg.lock]: This doesn't seem right - we want to have a gopkg.lock file

**Review by ecordell — CHANGES_REQUESTED** [review]

File: Gopkg.lock (hunk 1 above)
Comment (alecmerdler) [reply Gopkg.lock]: Fixed

**Comment by ecordell** [comment]
LGTM

**Comment by alecmerdler** [comment]
Don't know how to fix this:
```
[alec@daedalus lostromos]$ make out/lostromos-linux-amd64
Building for linux...
# github.com/wpengine/lostromos/vendor/k8s.io/client-go/plugin/pkg/client/auth/azure
vendor/k8s.io/client-go/plugin/pkg/client/auth/azure/azure.go:300:8: cannot use spt.Token (type func() adal.Token) as type adal.Token in field value
make: *** [Makefile:41: out/lostromos-linux-amd64] Error 2
```

**Comment by ecordell** [comment]
Built fine on my machine

**Review by ecordell — DISMISSED** [review]
LGTM

**Comment by alecmerdler** [comment]
Failing to build locally and using `Dockerfile`:
```
[alec@daedalus lostromos]$ make out/lostromos-linux-amd64
Building for linux...
# github.com/wpengine/lostromos/vendor/k8s.io/client-go/plugin/pkg/client/auth/azure
vendor/k8s.io/client-go/plugin/pkg/client/auth/azure/azure.go:300:8: cannot use spt.Token (type func() adal.Token) as type adal.Token in field value
make: *** [Makefile:41: out/lostromos-linux-amd64] Error 2
```

**Comment by alecmerdler** [comment]
Found the issue: https://github.com/coreos/lostromos/blob/master/Gopkg.lock#L28

**Review by ecordell — APPROVED** [review]
LGTM

**Pull request merged**